generator/lib/src/realm_field_info.dart

```diff
@@ -80,7 +80,13 @@ class RealmFieldInfo {
   Iterable<String> toCode() sync* {
     final getTypeName = type.isRealmCollection ? basicMappedTypeName : basicNonNullableMappedTypeName;
     yield '@override';
-    yield "$mappedTypeName get $name => RealmObjectBase.get<$getTypeName>(this, '$realmName') as $mappedTypeName;";
+    if (isRealmBacklink) {
+      yield "$mappedTypeName get $name {";
+      yield "if(!isManaged){throw RealmException('Using backlinks is only possible for managed(persisted) objects.');}";
```

Comment:
Should this be a `RealmError`. I think we are pretty incosistent with these validations but the question here is still valid I think?